src/background/feeder/e2e-session.feeder.ts

```diff
@@ -0,0 +1,194 @@
+      ) => {
+        const { action } = request || {}
+        const senderId = sender.id as string
+        const response: Response = {}
+
+        try {
+          if (!senderId) {
+            throw new Error('Invalid extension ID')
+          }
+
+          if (action === Action.REGISTER) {
+            response.data = this.handleRegistration(sender)
+          } else if (action === Action.LOCAL_STORAGE_GET) {
+            response.data = await this.handleLocalStorageGet(request)
+          } else if (action === Action.LOCAL_STORAGE_SET) {
+            await this.handleLocalStorageSet(request)
+          } else if (action === Action.WEB2_HELPER_FAKE_BEE_API_ADDRESS) {
+            response.data = this.handleWeb2FakeBeeApiAddress(request)
+          } else if (action === Action.WEB2_HELPER_FAKE_BZZ_ADDRESS) {
+            response.data = this.handleWeb2FakeBzzAddress(request)
+          } else if (action === Action.BZZ_LINK_PROTOCOL_TO_FAKE_URL) {
+            response.data = this.handleBzzLinkProtocolToFakeUrl(request)
+          } else if (action === Action.BZZ_LINK_LINK_URL_TO_FAKE_URL) {
+            response.data = this.handleBzzLinkUrlToFakeUrl(request)
+          } else if (action === Action.BZZ_LINK_URL_TO_FAKE_URL) {
+            response.data = this.handleBzzUrlToFakeUrl(request)
+          } else {
+            throw new Error(`Unknown action ${action}`)
+          }
+        } catch (error) {
+          response.error = String(error)
+        }
+
+        this.sendResponse(response, action, senderId, sendResponse)
+      },
+    )
+  }
+
+  private sendResponse(
+    response: Response | null,
+    action: Action,
+    senderId: string,
+    sendResponse: (response?: Response) => void,
+  ) {
+    if (response?.error) {
+      console.warn(
+        `Sending error response for action '${action}' to the extension with ID '${senderId}'.`,
+        response.error,
+      )
+    } else {
+      console.log(`The extension with ID '${senderId}' successfully invoked action '${action}'.`)
+    }
+    sendResponse(response || {})
+  }
+
+  private handleRegistration(sender: chrome.runtime.MessageSender): string {
+    const sessionId = nanoid()
+    this.manager.register(sessionId, sender)
+
+    return sessionId
+  }
+
+  private handleLocalStorageGet(request: RequestType): Promise<unknown> {
+    const {
+      sessionId,
+      parameters: { name },
+    } = request as LocalStorageGetRequest
+
+    if (!name) {
+      throw new Error('Cannot get item from localStorage. Item name required')
+    }
+
+    return this.manager.getStorageItem(sessionId, name)
+  }
+
+  private handleLocalStorageSet(request: RequestType): Promise<void> {
+    const {
+      sessionId,
+      parameters: { name, value },
+    } = request as LocalStorageSetRequest
+
+    if (!name) {
+      throw new Error('Cannot get item from localStorage. Item name required')
+    }
+
+    return this.manager.setStorageItem(sessionId, name, value)
+  }
```

Comment:
I'm not sure these functions have legitimacy as the extensions have their own local storage.

What use-cases you see for these features?

Reply:
Yes, there is no need to have these functions. I added them just to replicate the existing API. But for extensions there is no sense to have it. I've deleted them.